darts/ad/scorers.py

```diff
@@ -0,0 +1,1582 @@
+
+    def _diff(self, series_1: TimeSeries, series_2: TimeSeries) -> TimeSeries:
+        """Applies the reduced_function to the two time series. Converts two time series into 1.
+
+        series_1 and series_2 must be of the same length L and width/dimension W.
+
+        Parameters
+        ----------
+        series_1
+            1st time series
+        series_2:
+            2nd time series
+
+        Returns
+        -------
+        TimeSeries
+            series of length L and width/dimension W.
+        """
+        series_1, series_2 = _return_intersect(series_1, series_2)
+
+        if self.reduced_function == "abs_diff":
+            return (series_1 - series_2).map(lambda x: np.abs(x))
+        elif self.reduced_function == "diff":
+            return series_1 - series_2
+        else:
+            # found an non-existent reduced_function
+            raise ValueError(
+                f"Metric should be 'diff' or 'abs_diff', found {self.reduced_function}"
+            )
+
+
+# FittableAnomalyScorer
+
+
+class GaussianMixtureScorer(FittableAnomalyScorer):
+    """GaussianMixtureScorer anomaly score
+
+    Wrapped around the GaussianMixture scikit-learn function.
+    Source code: <https://github.com/scikit-learn/scikit-learn/blob/f3f51f9b6/sklearn/mixture/_gaussian_mixture.py#L456>
+    """
+
+    def __init__(
+        self,
+        window: Optional[int] = None,
+        n_components: int = 1,
+        reduced_function=None,
+        component_wise: bool = False,
+    ) -> None:
+
+        """
+        A Gaussian mixture model is trained on the training data when the ``fit()`` method is called.
+        The ``score()`` method will compute the log-likelihood of each sample.
+
+        TODO: stride in training is equal to w, and in score stride is equal to 1. Give the option to change
+        these parameters.
+
+        If 2 time series are given in the ``fit()`` or ``score()`` methods, a reduced function, given as a parameter
+        in the __init__ method (reduced_function), will be applied to transform the 2 time series into 1.
+        Default: "abs_diff"
+
+        component_wise is a boolean parameter in the __init__ method indicating how the model should behave with input
+        that is a multivariate series. If set to True, the model will treat each width/dimension of the series
+        independently. If the series has a width of d, the model will train and store the d Gaussian mixture models
+        and fit them on each dimension. If set to False, the model will concatenate the widths in the considered window
+        and compute the score using only one trained Gaussian mixture model.
+
+        Training:
+
+        The input can be a series (univariate or multivariate) or a list of series. The series will be partitioned
+        into equal size subsequences. If the series is multivariate (width>1), then the subsequence will be of size
+        window * width, with window being a given parameter.
```

Comment:
Here you use "window" and later you define "w" as a parameter. I would recommend you introduce all your notation upfront (e.g. saying say the input is a multivariate series of length `n`, dimension `d`, and we apply a window of size `w`, then ....).
Also use single bactickts for the variable names, it'll be clearer when the doc is rendered.